packages/devtools_app/lib/src/error_badge_manager.dart

```diff
@@ -66,15 +101,84 @@ class ErrorBadgeManager extends DisposableController
     notifier.value = currentCount + 1;
   }
 
+  void appendError(String screenId, DevToolsError error) {
+    final errors = _activeErrors[screenId];
+    if (errors == null) return;
+
+    // Build a new map with the new error. Adding to the existing map
+    // won't cause the ValueNotifier to fire (and it's not permitted to call
+    // notifyListeners() directly).
+    final newValue = LinkedHashMap<String, DevToolsError>.from(errors.value);
+    newValue[error.id] = error;
+    errors.value = newValue;
+    _updateErrorCount(screenId);
+  }
+
+  /// Update the error count to include only "unread" errors.
+  void _updateErrorCount(String screenId) {
+    final errors =
+        _activeErrors[screenId]?.value ?? const <String, DevToolsError>{};
+    _errorCountNotifier(screenId).value =
+        errors.values.where((err) => !err.read).length;
+  }
+
   ValueListenable<int> errorCountNotifier(String screenId) {
     return _errorCountNotifier(screenId) ?? const FixedValueListenable<int>(0);
   }
 
+  ValueListenable<LinkedHashMap<String, DevToolsError>> erroredWidgetNotifier(
```

Comment:
let's rename this to erroredElementNotifier, because "widget" is specific to the inspector screen, whereas the logging screen for example would be looking for errored "logs".  "element" seems like a good generic name.

Reply:
Element has a specific name in Flutter so I would avoid using it unless you want to imply something is a Flutter Element.

Reply:
Maybe erroredItemsForPage instead then

Reply:
Done in e74b35db